sklearn/utils/_repr_html/params.py

```diff
@@ -0,0 +1,83 @@
+# Authors: The scikit-learn developers
+# SPDX-License-Identifier: BSD-3-Clause
+
+import html
+import reprlib
+from collections import UserDict
+
+from sklearn.utils._repr_html.base import ReprHTMLMixin
+
+
+def _params_html_repr(params):
+    """Generate HTML representation of estimator parameters.
+
+    Creates an HTML table with parameter names and values, wrapped in a
+    collapsible details element. Parameters are styled differently based
+    on whether they are default or user-set values.
+    """
+    HTML_TEMPLATE = """
+        <div class="estimator-table">
+            <details>
+                <summary>Parameters</summary>
+                <table class="parameters-table">
+                  <tbody>
+                    {rows}
+                  </tbody>
+                </table>
+            </details>
+        </div>
+    """
+    ROW_TEMPLATE = """
+        <tr class="{param_type}">
+            <td><i class="fa-regular fa-copy"
+                 onclick="copyToClipboard('{param_name}',
+                          this.parentElement.nextElementSibling)"
+            ></i></td>
+            <td class="param">{param_name}&nbsp;</td>
+            <td class="value">{param_value}</td>
+        </tr>
+    """
+
+    rows = [
+        ROW_TEMPLATE.format(**_read_params(name, value, params.non_default))
+        for name, value in params.items()
+    ]
+
+    return HTML_TEMPLATE.format(rows="\n".join(rows))
+
+
+class ParamsDict(ReprHTMLMixin, UserDict):
+    """Dictionary-like class to store and provide an HTML representation.
+
+    It builds an HTML structure to be used with Jupyter notebooks or similar
+    environments. It allows storing metadata to track non-default parameters.
+
+    Parameters
+    ----------
+    params : dict, default=None
+        The original dictionary of parameters and their values.
+
+    non_default : tuple
+        The list of non-default parameters.
+    """
+
+    _html_repr = _params_html_repr
+
+    def __init__(self, params=None, non_default=tuple()):
+        super().__init__(params or {})
+        self.non_default = non_default
+
+
+def _read_params(name, value, non_default_params):
```

Comment:
Let's move this function on the top of the file since it already called in the first function.